lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -800,14 +800,15 @@ def create_autoscaling_group(connection, module):
 
         # Update load balancers if they are specified and one or more already exists
         elif as_group['LoadBalancerNames']:
+            change_load_balancers = load_balancers is not None
             # Get differences
             if not load_balancers:
                 load_balancers = list()
             wanted_elbs = set(load_balancers)
 
             has_elbs = set(as_group['LoadBalancerNames'])
             # check if all requested are already existing
-            if has_elbs.issuperset(wanted_elbs):
+            if has_elbs.issuperset(wanted_elbs) and change_load_balancers:
```

Comment:
If wanted_elbs is A and B, and has_elbs is B and C, has_elbs is not a superset, but C should be detached.

```
if has_elbs - wanted_elbs and change_load_balancers:
```

Or

```
if change_load_balancers:
    changed = True
    connection.detach_load_balancers(AutoScalingGroupName=group_name,
                                     LoadBalancerNames=list(has_elbs - wanted_elbs))
```